picard/ui/options/interface.py

```diff
@@ -227,16 +240,16 @@ def save(self):
         config.setting["use_adv_search_syntax"] = self.ui.use_adv_search_syntax.isChecked()
         config.setting["quit_confirmation"] = self.ui.quit_confirmation.isChecked()
         self.tagger.window.update_toolbar_style()
-        new_theme_setting = self.ui.use_system_theme.isChecked()
+        new_theme_setting = theme_enum_to_name(self.ui.ui_theme.itemData(self.ui.ui_theme.currentIndex()))
         new_language = self.ui.ui_language.itemData(self.ui.ui_language.currentIndex())
         restart_warning = None
-        if new_theme_setting != config.setting["use_system_theme"]:
+        if new_theme_setting != config.setting["ui_theme"]:
             restart_warning_title = _('Theme changed')
             restart_warning = _('You have changed the application theme. You have to restart Picard in order for the change to take effect.')
-            if new_theme_setting:
+            if new_theme_setting == UiTheme.SYSTEM:
```

Comment:
Here we compare result of `theme_enum_to_name()` and enum value (int).

Reply:
You're reviewing this way too fast @__@